src/mlpack/core/data/load.hpp

```diff
@@ -102,91 +105,12 @@ bool Load(const std::string& filename,
  * @param transpose If true, transpose the matrix after loading (default true).
  * @return Boolean value indicating success or failure of load.
  */
-template<typename eT>
+template<typename MatType>
 bool Load(const std::string& filename,
-          arma::SpMat<eT>& matrix,
+          arma::sp_mat& matrix,
           const bool fatal = false,
           const bool transpose = true);
 
```

Comment:
We need to handle this, I will try not to forget.

Reply:
If I am not mistaken, even if you have `SpMat<eT>` here, when you call `data::Load()`, the compiler will try to match the most specific version.  So calling `data::Load(filename, sparse_matrix, ...)` for some `sparse_matrix` that has type `SpMat<double>` or similar should work fine... I think.  But, I could be wrong---I am only writing this comment from memory, and I haven't actually tried this change in the code here.